rest/rest-server/src/main/java/org/apache/brooklyn/rest/resources/ApplicationResource.java

```diff
@@ -266,27 +268,48 @@ protected Response createFromAppSpec(ApplicationSpec applicationSpec) {
     @Override
     public Response createFromYaml(String yaml) {
         // First of all, see if it's a URL
-        URI uri;
+        Preconditions.checkNotNull(yaml, "Blueprint cannot be null");
+        URI uri = null;
         try {
-            uri = new URI(yaml);
+            String yamlUrl = yaml.trim();
+            if (Urls.isUrlWithProtocol(yamlUrl)) {
+                uri = new URI(yamlUrl);
+            }
         } catch (URISyntaxException e) {
             // It's not a URI then...
             uri = null;
         }
         if (uri != null) {
             log.debug("Create app called with URI; retrieving contents: {}", uri);
-            yaml = ResourceUtils.create(mgmt()).getResourceAsString(uri.toString());
+            try {
+                yaml = ResourceUtils.create(mgmt()).getResourceAsString(uri.toString());
+            } catch (Exception e) {
+                Exceptions.propagateIfFatal(e);
+                throw new UserFacingException("Cannot resolve URL: "+uri, e);
+            }
         }
 
         log.debug("Creating app from yaml:\n{}", yaml);
-        EntitySpec<? extends Application> spec = createEntitySpecForApplication(yaml);
+        EntitySpec<? extends Application> spec;
+        try {
+            spec = createEntitySpecForApplication(yaml);
+        } catch (Exception e) {
+            Exceptions.propagateIfFatal(e);
+            Throwable root = Throwables.getRootCause(e);
+            if (root instanceof UserFacingException) throw (UserFacingException) root;
+            throw WebResourceUtils.badRequest(e, "Error in blueprint");
+        }
         
         if (!Entitlements.isEntitled(mgmt().getEntitlementManager(), Entitlements.DEPLOY_APPLICATION, spec)) {
             throw WebResourceUtils.unauthorized("User '%s' is not authorized to start application %s",
                 Entitlements.getEntitlementContext().user(), yaml);
         }
 
-        return launch(yaml, spec);
+        try {
+            return launch(yaml, spec);
+        } catch (Exception e) {
+            throw WebResourceUtils.badRequest(e, "Error launching blueprint");
```

Comment:
not sure if it's used in launch, unless internally in an application ... FWIW there is some similar `UserFacing` preference in `DefaultExceptionMapper` (which I've updated similarly) -- let's see how this works and tweak it or overhaul it as we discover we need to